### PR TITLE
dvdplayer fix sorting of subtitles - fixes trac 16038

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
good for merge or need confirmation from ticket ?
jenkins build this please

**Comment by FernetMenta** [comment]
I tried with the sample ass file provided by the ticket owner and it did work. Does not harm if someone like @Paxxi has an additional look on this. Note that this a quick fix for I*. The entire class should be rewritten in order to make use of standard collections and sorting methods.

File: xbmc/cores/dvdplayer/DVDSubtitles/DVDSubtitleLineCollection.cpp [hunk 1]
```diff
@@ -60,9 +60,9 @@ void CDVDSubtitleLineCollection::Sort()
   if (!m_pHead || !m_pHead->pNext)
     return;
   
-  for (ListElement* p1 = m_pHead; p1->pNext->pNext != NULL; p1 = p1->pNext)
+  for (ListElement* p1 = m_pHead; p1->pNext != NULL; p1 = p1->pNext)
```
Comment (ace20022) [inline xbmc/cores/dvdplayer/DVDSubtitles/DVDSubtitleLineCollection.cpp]: Shouldn't this also check p1?

Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDSubtitles/DVDSubtitleLineCollection.cpp]: 2 lines above it checks for m_pHead which is p1 at starting point

Comment (ace20022) [inline xbmc/cores/dvdplayer/DVDSubtitles/DVDSubtitleLineCollection.cpp]: yes, sorry for the noise.

**Comment by Paxxi** [comment]
I can't see anything wrong here, should be good to go codewise.
agree with @FernetMenta it really should get rewritten after Isengard

**Pull request merged**